aptos-move/framework/move-stdlib/sources/configs/config_for_next_epoch.move

```diff
@@ -0,0 +1,126 @@
+/// This wrapper helps store an on-chain config for the next epoch.
+module std::config_for_next_epoch {
+
+    use std::signer::address_of;
+
+    const ESTD_OR_VM_SIGNER_NEEDED: u64 = 1;
+    const ESTD_SIGNER_NEEDED: u64 = 2;
+    const EVM_SIGNER_NEEDED: u64 = 3;
+    const ERESOURCE_BUSY: u64 = 4;
+    const EPERMISSION_DENIED: u64 = 5;
+
+    /// `0x1::ForNextEpoch<T>` holds the config payload for the next epoch, where `T` can be `ConsnsusConfig`, `Features`, etc.
+    struct ForNextEpoch<T> has drop, key {
+        payload: T,
+    }
+
+    /// We need to temporarily reject on-chain config changes during DKG.
+    /// `0x0::UpdateLock` or `0x1::UpdateLock`, whichever has the higher `seq_num`, represents whether we should reject.
+    struct UpsertLock has copy, drop, key {
+        seq_num: u64,
+        locked: bool,
+    }
```

Comment:
This comes from the requested behavior: when a slow reconfigure is in progress, on-chain config changes should be rejected.

Ideally we define a struct `T` to represent the lock and put a `0x1::T` when starting DKG and remove it when finishing. But reconfigure can be triggered by either 0x1 (proposal) or 0x0 (epoch expiry), and 0x0 can't put things under 0x1. 2 solutions i can imagine.

1. Define `struct SomeLock { locked: bool }`; run a initialization script to put `0x1::SomeLock`. Later both 0x0 and 0x1 can read/write the `locked` field.
2. (implemented currently in the PR) Have 0x0 and 0x1 maintain a struct under their own address. Each maintain a `seqnum` so we know who is the last write. When reading the lock state, read from the last write.